core/parse.y

```diff
@@ -255,24 +255,28 @@ expr:
 	}
 |	'(' expr for_clauses if_clauses_opt ')'
 	{
+		exprStart, _ := $2.Span()
 		$$ = &ListForExpr{
 			Brack: "()",
 			Start: $1,
 			X: $2,
 			For: $3,
 			If: $4,
 			End: End{Pos: $5},
+			ForceMultiLine: $1.Line != exprStart.Line,
```

Comment:
should this also have a line-length component?  If I have a short but wrapped line I would like it to be made smaller
[
do_this
for f in files
]

--- should become ---
[do_this for f in files]